### PR TITLE
add github workflow

### PR DESCRIPTION
**Pull request by iamchrissmith**

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  # No optimizations
+  SOLC_FLAGS = "";
+  buildInputs = [
+    dapp
+  ];
+
+  shellHook = ''
+    export NIX_SSL_CERT_FILE=${cacert}/etc/ssl/certs/ca-bundle.crt
+    unset SSL_CERT_FILE
+
```
Comment (naszam) [inline shell.nix]: nit: extra line

**Review by naszam — APPROVED** [review]
LGTM

**Review by julienmartinlevrai — APPROVED** [review]
Diffed with https://github.com/makerdao/dss-direct-deposit, LGTM

Note: this repo uses Solidity 0.8.0. If we need 0.8, I find it advisable to use 0.8.12 or later to have as many bugs fixed as possible.

**Comment by iamchrissmith** [comment]
> Diffed with https://github.com/makerdao/dss-direct-deposit, LGTM
> 
> Note: this repo uses Solidity 0.8.0. If we need 0.8, I find it advisable to use 0.8.12 or later to have as many bugs fixed as possible.

Yeah i wanted to keep this one to just the ci stuff, but will update the version to our standard new one

**Pull request merged**